### DISCUSSION TITLE
Welcome to BattMo discussions!

### DISCUSSION THREAD

**Discussion by jsimonclark** [BattMoTeam/BattMo#10] [Announcements]
Welcome battery modellers! 

We are starting a discussion board to help BattMo users and developers connect with and support each other. We invite you to use this space to discuss your questions and ideas for further development. 

Happy modelling!

**Comment by anirbanroy16** [comment]
Hello fellow modelers!

Glad to meet y'all. 

My name is Anirban Roy and I am a data analyst and modeler with Ambri, a liquid-metal battery startup based outside the Boston area. Happy modeling :-)

**Comment by anirbanroy16** [comment]
I did notice the temperature-dependence of open circuit potential is not included - we could check the uncertainty in predictions by incorporating the same in. Here's an example modeling effort (LION-SIMBA) which parameterizes the OCP T-dependence of graphite in Table II:  https://web.mit.edu/braatzgroup/Torchio_JElectSoc_2016.pdf

**Reply by jsimonclark** [reply]
The temperature-dependence of the OCP is included in **_some_** parameter sets, but not all. As you point out, the Torchio2016 parameter set includes it. Also Sturm 2019. Chen 2020 does not, but they did an update with the temperature dependence in 2022. We're working to make the selection of parameter sets - and what is and is not included in each - more clear in the upcoming release. For reference, here is an example of how a temperature-dependent OCP is handled:

https://github.com/BattMoTeam/BattMo/blob/main/ParameterData/MaterialProperties/NMC/computeOCP_nmc111.m

**Comment by anirbanroy16** [comment]
Apropos the file mrstModule.m, what does this specifically mean?

  * add - Activate specified modules from the [module list].

**Reply by augustjohansson** [reply]
BattMo makes use of grids, time stepping, optimization tools and many other core components from MRST, a reservoir simulation software. MRST is based upon modules which you load using `mrstModule add`. Basically what this does is adding directories to the path.

**Comment by anirbanroy16** [comment]
Also, why is it necessary to run startupBattMo before running any example?

**Reply by augustjohansson** [reply]
`startupBattMo` setup the paths to the BattMo source code and some paths so we can use certain modules from MRST. It should not be required to run `startupBattMo` every time you run an example. It should be sufficient to start matlab and run `startupBattMo` once. Then you should be able to use all of BattMo!

**Reply by anirbanroy16** [reply]
Great, thanks August!

**Comment by anirbanroy16** [comment]
What is the difference between a simple and full diffusion model?

**Reply by augustjohansson** [reply]
The diffusion of lithium in the solid electrodes is typically modeled by introducing solid particles in the electrodes. The diffusion of the lithium in the electrodes is then modeled by the diffusion in these solid particles. These local diffusion problems (which is mathematically described by the Fick's law and the conservation equation, i.e. PDEs) can then be solved as we solve any other PDE: by discretization in time and space. This is what is done in the so-called full diffusion model. However, we make also make use of approximate analytical solutions, which are faster but less accurate. In the simple diffusion model we have a quadratic polynomial as an approximation. All this is described in the Zhang & White paper https://doi.org/10.1016/j.jpowsour.2006.12.056.

The idea of modeling the diffusion using particles is what makes this Li-ion battery model known as the pseudo 2D (P2D) model, since you have local problems that doesn't live in physical space. The same holds then for P3D and P4D models.

**Reply by anirbanroy16** [reply]
Great, thanks!

So basically a single particle model, with an electrode particle discretized into "onion slices" and solved using a finite volume method, is an example of "full diffusion" - is that correct?

Now I remember these options in the LION-SIMBA model as well.

**Reply by augustjohansson** [reply]
Yes, that is correct. I think many major battery simulation frameworks have this type of implementation since this is pretty standard.

**Comment by aykharal** [comment]
How can I fit the parameters on experimental data?

**Comment by anirbanroy16** [comment]
Hi,

I'm not sure how to do this with BattMo but there is a Simulink example to
do this for physics-based models

https://www.mathworks.com/help/sldo/ug/perform-grouped-estimation-of-model-parameters-for-single-particle-battery-model.html

If you're looking at tuning equivalent circuit parameters, there are a
litany of examples.on how to estimate these too.

https://www.mathworks.com/videos/estimating-parameters-of-a-battery-68957.html

But please just keep mind the approaches to estimate the cost function
(e.g. simplex algorithm) often do not reach the local minimum so you'd need
to start with a grid. However, genetic algorithms are better suited to
predict the global minimum.

Finally, the fitted parameters as a function of temperature and SoC can
fluctuate wildly due to overflowing. If possible, use the model calibration
toolbox in lieu of parameter estimator.

All the best!

Sincerely,
Anirban

**Reply by aykharal** [reply]
Thanks, I'll look into this. Actually i have been trying to use BattMo and COBRA PRO (developed by stanford's SLAC group), and PyBamm as well. All models run good on the examples provided but hard to fit on the experimental data. 

ECM fitting is not an issue as i have estimated ECM parameters using EIS fitting. 
Thanks for your kind suggestions.

**Comment by anirbanroy16** [comment]
Yes this is typically the case, these parameters have to be tuned to
specific conditions, especially if the optimizer is very sensitive to the
first guess.
